### PR TITLE
database/raft: remove debugging log messages

### PR DESCRIPTION
**Pull request by kr**
We want cored log output to be useful for operators. For
our own debugging purposes, we can add and remove
logging as necessary during development, or use other
tools such as memory debuggers and tracing.

### CODE REVIEW

File: database/raft/raft.go [hunk 1]
```diff
@@ -607,7 +602,7 @@ func (sv *Service) serveJoin(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	log.Printkv(req.Context(), "at", "join-done", "addr", x.Addr, "id", newID)
+	log.Printkv(req.Context(), "at", "serveJoin", "addr", x.Addr, "id", newID)
```
Comment (kr) [inline database/raft/raft.go]: Not sure if we should keep this one. (Or maybe reword it.)

File: database/raft/raft.go [hunk 2]
```diff
@@ -650,17 +645,15 @@ func (sv *Service) join(addr, baseURL string) error {
...
+	if !raft.IsEmptySnap(raftSnap) {
 		err = sv.state.RestoreSnapshot(raftSnap.Data, raftSnap.Metadata.Index)
 		if err != nil {
 			return errors.Wrap(err)
 		}
 		sv.confState = raftSnap.Metadata.ConfState
 		sv.snapIndex = raftSnap.Metadata.Index
-		log.Printkv(ctx, "at", "joined", "appliedindex", raftSnap.Metadata.Index)
 	}
+	log.Printkv(ctx, "at", "joined", "appliedindex", raftSnap.Metadata.Index)
```
Comment (kr) [inline database/raft/raft.go]: Also not sure if we should keep this one.

File: database/raft/raft.go [hunk 3]
```diff
@@ -201,6 +202,7 @@ func Start(laddr, dir, bootURL string, httpClient *http.Client) (*Service, error
 		Applied:         sv.state.AppliedIndex(),
 		MaxSizePerMsg:   4096,
 		MaxInflightMsgs: 256,
+		Logger:          &raft.DefaultLogger{Logger: stdlog.New(ioutil.Discard, "", 0)},
```
Comment (tessr) [inline database/raft/raft.go]: what does this line do?

File: database/raft/raft.go [hunk 4]
```diff
@@ -510,7 +510,6 @@ func (sv *Service) get(ctx context.Context, key string) ([]byte, error) {
 			err := sv.Set(ctx, "/dummyWrite", []byte(""))
 			if err != nil {
 				// TODO(tessr): should probably return this error
```
Comment (tessr) [inline database/raft/raft.go]: could you update this todo to "do something about this error"? 

or... not? we do occasionally get this error and it's totally fine, the node just retries the dummy write.

File: database/raft/raft.go (hunk 2 above)
Comment (tessr) [inline database/raft/raft.go]: I think one "joined the cluster" message is useful, probably don't need both of them.

Comment (kr) [inline database/raft/raft.go]: Okay, then any opinion on which one we should keep (new member vs existing member)? I will prob remove the serveJoin message since it would happen on an arbitrary existing member. This one on the new member is prob more salient.

File: database/raft/raft.go (hunk 3 above)
Comment (kr) [inline database/raft/raft.go]: It discards the coreos raft package log output. Its [default logger](https://github.com/chain/chain/blob/50dff7d598be223161a5a56299719cff95cd99ab/vendor/github.com/coreos/etcd/raft/logger.go#L47-L49) goes to stderr.

I didn't see any difference in the output in my tests, but I suppose it's possible I just didn't hit anything that causes that package to print log messages.

File: database/raft/raft.go (hunk 4 above)
Comment (kr) [inline database/raft/raft.go]: Aha okay, I'll remove the todo (and add a comment saying it's okay to ignore the error).

File: database/raft/raft.go (hunk 3 above)
Comment (tessr) [inline database/raft/raft.go]: So if we're trying to debug, we can just pass in a different io writer? SGTM

File: database/raft/raft.go (hunk 2 above)
Comment (tessr) [inline database/raft/raft.go]: SGTM

**Comment by tessr** [comment]
[approved]
